pkg/templates/generic_lookup_func.go

```diff
@@ -152,6 +161,18 @@ func (t *TemplateResolver) getDynamicClient(
 	}
 	klog.V(2).Infof("GVR is:  %v", gvr)
 
+	if !apiResource.Namespaced && t.config.LookupNamespace != "" {
+		rsrcIdentifier := ClusterScopedObjectIdentifier{
+			Group:   apiResource.Group,
+			Version: apiResource.Version,
```

Comment:
I don't think the version should be used in the restriction. For instance, if ClusterClaim has an API version bump, I wouldn't want the calling code to be updated.